test-suites/helpers/validated-steps.ts

```diff
@@ -567,22 +595,258 @@ const checkBeforeLiquidation = async (before: LiquidationValidationData) => {
   }
 };
 
-// refactor: move check logics here before and after liquidation
-/* eslint-disable  no-unused-vars */
 const checkAfterLiquidationERC20 = async (
-  before: LiquidationValidationData,
-  after: LiquidationValidationData
+  borrower: SignerWithAddress,
+  targetToken: SupportedAsset,
+  liquidationToken: SupportedAsset,
+  amountToLiquidate: BigNumber,
+  receivePToken: boolean,
+  before: BeforeLiquidationValidationData,
+  after: AfterLiquidationValidationData
 ) => {
-  return true;
+  // borrower's Token balance is the same
+  expect(after.borrowerTargetTokenBalance).equal(
+    before.borrowerTargetTokenBalance
+  );
+
+  // borrower's pToken balance is subtracted amountToLiquidate+liq bonus (bonus is paid by borrower)
+  assertAlmostEqual(
+    after.borrowerTargetXTokenBalance,
+    before.borrowerTargetXTokenBalance.sub(
+      amountToLiquidate.mul(before.liquidationBonus).div(10000)
+    )
+  );
+
+  // borrower debtToken balance is subtracted amountToLiquidate (plus accrued interest)
+  assertAlmostEqual(
+    after.borrowerLiquidationDebtTokenBalance,
+    before.borrowerLiquidationDebtTokenBalance.sub(amountToLiquidate)
+  );
+
+  if (receivePToken) {
+    // liquidator's Token balance is subtracted the liquidated amount
+    assertAlmostEqual(
+      after.liquidatorTargetTokenBalance,
+      before.liquidatorTargetTokenBalance.sub(amountToLiquidate)
+    );
+    // liquidator's pToken balance is incremented in liquidatedAmount+bonus (plus accrued interest)
+    assertAlmostEqual(
+      after.liquidatorTargetXTokenBalance,
+      before.liquidatorTargetXTokenBalance.add(
+        amountToLiquidate.mul(before.liquidationBonus).div(10000)
+      )
+    );
```

Comment:
also these checks will work?

Reply:
would suggest we make sure every check make sense if in a common function to reuse. or we just wrap result including enough info and let test verify by itself.